app/models/client_contacts.rb

```diff
@@ -0,0 +1,239 @@
+class ClientContacts
```

Comment:
This class is fine (duplicate of ProgramContacts) but it might make sense to pull the reusable bits into a concern or a parent class.  I believe many of the methods are identical and the `label_for`, `contact_type_for`, etc. seem like they might diverge accidentally eventually.

Reply:
There are many places where this is true. It is in match_contacts also.   Based on conversations yesterday, I had assumed that refactoring is not on the plate, but if you want all of the common code combined, I can certainly do that

Reply:
I wouldn't go and refactor other code duplication, but since this was an N of one, and is now and N of two, it seems like a good time to avoid the duplication spreading.